lib/vrl/compiler/src/compiler.rs

```diff
@@ -204,26 +204,69 @@ impl<'a> Compiler<'a> {
         })
     }
 
+    /// Rewrites the ast for `a |= b` to be `a = a | b`.
+    fn rewrite_to_merge(
+        &mut self,
+        span: diagnostic::Span,
+        target: &Node<ast::AssignmentTarget>,
+        expr: Box<Node<ast::Expr>>,
+    ) -> Box<Node<Expr>> {
```

Comment:
Nit: I'd let the caller box this one

Reply:
For what reason?

Reply:
Simply because this function shouldn't be concerned about where the result is allocated. Hypothetically, if you wanted to allocate the result on the stack, this would prevent you from doing it.

This is just an argument about principle, not about this practical case. Hence why only a nit and up to your discretion to change it or not 😄

Reply:
Ah yeah, that's a good point. I won't make that change here, just because it is only called from two places, and both places need to box it, but will bear that in mind for the future! Thanks.